monitoring/k8s/grafana/turbinia-application-metrics.yaml

```diff
@@ -0,0 +1,1110 @@
+apiVersion: v1
```

Comment:
This file looks like it's only about 10 lines of header/footer data, and the rest is the same as the corresponding .json file.   Just to reduce the amount of duplicate data, and to make it easier to keep things in sync, could we make this a template file, and then just run a sed command to include the other file?

One way to do it with sed is to have the template file:
```
apiVersion: v1
items:
- apiVersion: v1
  data:
      turbinia-application-metrics.json: |-
@@JSONDATA@@
  kind: ConfigMap
  metadata:
    labels:
      app.kubernetes.io/component: grafana
      app.kubernetes.io/name: grafana
      app.kubernetes.io/part-of: kube-prometheus
      app.kubernetes.io/version: 7.5.4
    name: turbinia-application-metrics
    namespace: monitoring
kind: ConfigMapList
```

And then a one-liner to replace the @@JSONDATA@@ with the other file:
```
sed -e '/@@JSONDATA@@/{r turbinia-application-metrics.json' -e 'd}' -i turbinia-application-metrics.yaml
```

Since there are a few files, we could also put the sed lines into short `gen-yaml.sh` script or something.  Then later when we have the updated version of `deploy.sh` we can just include those there.

Reply:
Agree with that approach.

Reply:
Added a script named `gen-yaml.sh` that automatically updates the application and healthcheck dashboards and the prometheus rules section